crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -284,49 +289,77 @@ impl SolverTimeout {
     }
 }
 
-/// The solution score. This is often referred to as the "objective value".
-#[derive(Debug, Default, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
-pub struct Score(pub eth::U256);
+/// Carries information how the score should be calculated.
+#[derive(Debug, Clone)]
+pub enum Score {
+    Solver(eth::U256),
+    Discount(eth::U256),
+    RiskAdjusted {
+        success_probability: SuccessProbability,
+        gas_amount: Option<eth::U256>,
```

Comment:
For the same reason we added it to the [http_solver](https://github.com/cowprotocol/services/blob/optimal_bidding-colocation/crates/shared/src/http_solver/model.rs#L214).
In case the solver provided the gas_amount here it will be used for objective value calculation in `driver`.

Reply:
Did you get a chance to see if we can make our gas amount estimation more accurate (e.g. by using a metering contract or a self hosted Enso trade simulator)? If so, I would revert the decision of letting solvers specify the gas amount (it's weird that on the one hand we don't allow simulation in the solver, yet we expect solvers to provide more accurate gas estimates than our driver can deliver) and fix the accuracy in a separate step.

Reply:
> Did you get a chance to see if we can make our gas amount estimation more accurate

Unfortunately not yet

Reply:
Removed `gas_amount` from driver.
Note that is still temporarily exists for `http_solver` for backward compatibility. 
Implementing accurate gas estimation is part of https://github.com/cowprotocol/services/issues/1893